### PR TITLE
Adjust OGL to run with ec2 github runner

### PR DESCRIPTION
**Pull request by HendriceH**
std::filesystem::create_directories is commentented out due to error during creation of the lib.
Might be due to mixup of some gcc7 (standard on instance) and gcc10 (post-installed) mixup.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -56,13 +56,15 @@ if(NOT ${OGL_USE_EXTERNAL_GINKGO})
...
     CONFIG
-    1.5.0
+    1.7.0
     REQUIRED
     PATHS
     $ENV{FOAM_USER_LIBBIN}/lib/cmake/Ginkgo
     ${CMAKE_CURRENT_BINARY_DIR}/third_party/ginkgo/build/install/lib/cmake)
 endif()
 
+find_package(CUDAToolkit REQUIRED)
```
Comment (greole) [inline CMakeLists.txt]: Can you move the find package to the if({GINKGO_BUILD_CUDA}) section
```suggestion
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -195,7 +197,7 @@ if(${GINKGO_WITH_OGL_EXTENSIONS})
 endif()
 if(${GINKGO_BUILD_CUDA})
```
Comment (greole) [inline CMakeLists.txt:197]:
```suggestion
if(${GINKGO_BUILD_CUDA})
  find_package(CUDAToolkit REQUIRED)
```

File: third_party/ginkgo/CMakeLists.txt [hunk 3]
```diff
@@ -12,6 +12,7 @@ ginkgo_load_git_package(
   "-DGINKGO_BUILD_BENCHMARKS=OFF"
   "-DGINKGO_BUILD_REFERENCE=${GINKGO_BUILD_REFERENCE}"
   "-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}"
+  "-DCMAKE_CUDA_ARCHITECTURES=75"
```
Comment (greole) [inline third_party/ginkgo/CMakeLists.txt]:
```suggestion
```
We don't want to hardcode the CUDA architecture. Does the automatic detection still fail?

File: common/common.C [hunk 4]
```diff
@@ -62,7 +62,7 @@ void export_mtx(const word fieldName, std::shared_ptr<const gko::LinOp> A,
                 const word matrixFormat)
 {
     std::string folder{db.time().timePath()};
-    std::filesystem::create_directories(folder);
+    //std::filesystem::create_directories(folder);
```
Comment (greole) [inline common/common.C]:
```suggestion
    std::filesystem::create_directories(folder);
```
For the integration tests to work we still need this. To fix the issue you can use 
`target_link_libraries(OGL PUBLIC Ginkgo::ginkgo stdc++fs)` in the root CMakeLists.txt.

**Review by greole — COMMENTED** [review]
Thanks alot. Here are some minor comments.

File: third_party/ginkgo/CMakeLists.txt (hunk 3 above)
Comment (HendriceH) [reply third_party/ginkgo/CMakeLists.txt]: I switched to a newer cmake and switched to "native".

**Comment by greole** [comment]
@HendriceH I changed to merge into [refact/sparse_comm](https://github.com/hpsim/OGL/tree/refact/sparse_comm) otherwise compilation fails because the dev branch pulls an older version of ginkgo

**Pull request merged**